bin/packer-build-script

```diff
@@ -1,12 +1,63 @@
 #!/usr/bin/env bash
-: "${AWS_ACCESS_KEY:=$AWS_ACCESS_KEY_ID}"
-: "${AWS_SECRET_KEY:=$AWS_SECRET_ACCESS_KEY}"
-export AWS_ACCESS_KEY_ID AWS_ACCESS_KEY
-export AWS_SECRET_ACCESS_KEY AWS_SECRET_KEY
-
 set -o errexit
 
-cd "$(dirname "$(dirname "${BASH_SOURCE[0]}")")"
+main() {
+  : "${AWS_ACCESS_KEY:=$AWS_ACCESS_KEY_ID}"
+  : "${AWS_SECRET_KEY:=$AWS_SECRET_ACCESS_KEY}"
+  export AWS_ACCESS_KEY_ID AWS_ACCESS_KEY
+  export AWS_SECRET_ACCESS_KEY AWS_SECRET_KEY
+
+  local template_name="${1}"
+  local pt
+  pt="$(__abspath "$(dirname "$(dirname "${BASH_SOURCE[0]}")")")"
+
+  if [[ ! "${DISABLE_PACKER_BUILD}" ]]; then
+    __packer_build "${pt}" "${template_name}"
+  fi
+
+  if [[ ! "${TRAVIS_BUILD_DIR}" ]]; then
+    exit 0
+  fi
+
+  echo "     Found \$TRAVIS_BUILD_DIR." >&2
+  echo "     Copying metadata tarball to ${TRAVIS_BUILD_DIR}." >&2
+  __copy_metadata_tarball "${pt}/tmp" "${TRAVIS_BUILD_DIR}" "${template_name}"
+}
+
+__packer_build() {
+  local pt="${1}"
+  local template_name="${2}"
+  pushd "${pt}"
+  make
+  make "${template_name}"
+  popd
+}
+
+__copy_metadata_tarball() {
+  local pttmp
+  pttmp="$(__abspath "${1}")"
+  local top
+  top="$(__abspath "${2}")"
+  local template_name="${3}"
+
+  local packer_build_image_name
+  packer_build_image_name="$(
+    cat "${pttmp}/${template_name}/latest-image-name" 2>/dev/null || echo empty
+  )"
+  touch "${pttmp}/image-metadata-${packer_build_image_name}.tar.bz2"
+  cp "${pttmp}/image-metadata-${packer_build_image_name}.tar.bz2" \
+    "${top}/image-metadata-${packer_build_image_name}.tar.bz2"
+
+  echo "     Writing ${packer_build_image_name} to:" >&2
+  echo "         ${top}/.packer-build-image-name" >&2
+  echo "${packer_build_image_name}" >"${top}/.packer-build-image-name"
+}
+
+__abspath() {
+  python <<EOPYTHON
+import os, sys
+sys.stdout.write(os.path.abspath("${1}") + "\\n")
+EOPYTHON
```

Comment:
@igorwwwwwwwwwwwwwwwwwwww I'm +0 to using `sys.argv` in there.  Maybe a future iteration if there are issues?